packages/features/ee/payments/components/Payment.tsx

```diff
@@ -66,7 +66,7 @@ export default function PaymentComponent(props: Props) {
   const handleSubmit = async (ev: SyntheticEvent) => {
     ev.preventDefault();
 
-    if (!stripe || !elements) return;
+    if (!stripe || !elements || !router.isReady) return;
```

Comment:
Since we're passing a URL param, ensure that the router is ready or else it'll come up as undefined